onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -917,3 +915,47 @@ TEST(CApiTest, TestSharedAllocatorUsingCreateAndRegisterAllocator) {
                     expected_values_y,
                     nullptr);
 }
+
+TEST(CApiTest, TestSharingOfInitializer) {
+  // simple inference test
+  // prepare inputs
+  std::vector<Input> inputs(1);
+  Input& input = inputs.back();
+  input.name = "X";
+  input.dims = {3, 2};
+  input.values = {1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f};
+
+  // prepare expected inputs and outputs
+  std::vector<int64_t> expected_dims_y = {3, 2};
+  std::vector<float> expected_values_y = {1.0f, 4.0f, 9.0f, 16.0f, 25.0f, 36.0f};
+
+  Ort::SessionOptions session_options;
+  Ort::MemoryInfo mem_info = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);
+  float data[] = {1., 2., 3., 4., 5., 6.};
+  const int data_len = sizeof(data) / sizeof(data[0]);
```

Comment:
>data_len [](start = 12, length = 8)

Do we have to calculate this since it is constant whatsoever.
And if we do, maybe we should let it be prefixed by constexpr.

Reply:
It's done at compile time.